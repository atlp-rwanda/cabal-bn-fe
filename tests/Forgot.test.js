import * as React from 'react';
import {
  render,
  screen,
  waitFor,
  fireEvent,
  act,
} from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { Provider, useSelector } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../src/redux/store';
import Forgot from '../src/view/Forgot';
import { sendEmail } from '../src/redux/actions/Forgot.action';

describe('FORGOT PASSWORD', () => {
  it('should renders forgot component', () => {
    const element = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Forgot />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(element).toMatchSnapshot();
  });
  it('should testing invalid empty Email', () => {
    render(
      <Provider store={store}>
        <Router>
          <Forgot />
        </Router>
      </Provider>,
    );
    const email = screen.getByLabelText('Email');
    fireEvent.change(email, { target: { value: ' ' } });
    const button = screen.getByRole('button', { name: 'Submit' });
    fireEvent.click(button);
    const validation = screen.findByText('Email required');
  });
  it('should testing valid Email', () => {
    render(
      <Provider store={store}>
        <Router>
          <Forgot />
        </Router>
      </Provider>,
    );
    const email = screen.getByLabelText('Email');
    fireEvent.change(email, { target: { value: 'REQUESTER@gmail.com' } });
    const button = screen.getByRole('button', { name: 'Submit' });
    fireEvent.click(button);
    const validation = screen.findByText(
      'Password reset link has been sent to your email',
    );
    expect(sendEmail(email));
  });
});
