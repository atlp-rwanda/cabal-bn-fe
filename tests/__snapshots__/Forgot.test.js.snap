// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FORGOT PASSWORD should renders forgot component 1`] = `
Array [
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1xbm663-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <div
        className="css-1a62bo9"
      >
        <img
          alt="logo"
          src="Group7.svg"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="css-c0aifp-MuiStack-root"
      >
        <a
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "#00095E",
              "textDecoration": "none",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-aztsm3-MuiTypography-root"
          >
            Home
          </p>
        </a>
        <a
          href="/#undefined"
          onClick={[Function]}
          style={
            Object {
              "color": "#00095E",
              "textDecoration": "none",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-aztsm3-MuiTypography-root"
          >
            About
          </p>
        </a>
        <a
          href="/Accommodation"
          onClick={[Function]}
          style={
            Object {
              "color": "#00095E",
              "textDecoration": "none",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-aztsm3-MuiTypography-root"
          >
            Accommodation
          </p>
        </a>
        <a
          href="/login"
          onClick={[Function]}
          style={
            Object {
              "color": "#00095E",
              "textDecoration": "none",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-aztsm3-MuiTypography-root"
          >
            Sign In
          </p>
        </a>
        <a
          href="/signup"
          onClick={[Function]}
          style={
            Object {
              "color": "#00095E",
              "textDecoration": "none",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-aztsm3-MuiTypography-root"
          >
            Sign Up
          </p>
        </a>
      </div>
      <div
        className="css-1agvr0s"
        onClick={[Function]}
      >
        <img
          alt="drop down"
          src="Group345.svg"
          style={
            Object {
              "height": "40px",
              "marginTop": "8px",
              "width": "130px",
            }
          }
        />
      </div>
    </div>
  </header>,
  <div
    className="MuiBox-root css-mhf6ze"
  >
    <div
      className="MuiBox-root css-rntpfg"
    >
       
      <img
        alt="reset"
        className="css-102cdgy"
        src="undraw_forgot_password_re_hxwm 1.svg"
      />
    </div>
    <div
      className="MuiBox-root css-1429cz4"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 css-3kp7dz-MuiTypography-root"
      >
         
        Forgot Password
      </h4>
      <div
        className="css-s3cggk-MuiStack-root"
      >
         
         
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root css-1x1ecdk-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r0:"
          id=":r0:-label"
        >
          Email
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r0:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                Email
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-1lp41e9-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
         
        Submit
      </button>
       
    </div>
  </div>,
  " ",
  <div
    className="css-1izbarm"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidth100% css-176flsv-MuiContainer-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container css-9x7x8m-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-5 MuiGrid-direction-xs-[object Object] css-1bfbvf5-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              className="css-1usxeao-MuiStack-root"
            >
              <img
                alt="logo"
                height="62px"
                src="footerLogo.svg"
                width="130.2px"
              />
              <p
                className="MuiTypography-root MuiTypography-body1 css-9wmnc6-MuiTypography-root"
              >
                Land behold it created good saw after she&aposd Our set living. Signs midst dominion creepeth morning laboris nisi ufsit aliquip.
              </p>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              className="css-1f85j5c-MuiStack-root"
            >
              <p
                className="MuiTypography-root MuiTypography-body1 css-s1fqaa-MuiTypography-root"
              >
                Navigation
              </p>
              <p
                className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
              >
                About Us
              </p>
              <p
                className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
              >
                Accommodation
              </p>
              <p
                className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
              >
                SignIn
              </p>
              <p
                className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
              >
                Sign Up
              </p>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              className="css-1f85j5c-MuiStack-root"
            >
              <p
                className="MuiTypography-root MuiTypography-body1 css-s1fqaa-MuiTypography-root"
              >
                Services
              </p>
              <p
                className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
              >
                Hotel Booking
              </p>
              <p
                className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
              >
                Trip Booking
              </p>
              <p
                className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
              >
                Destination Booking
              </p>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-s1fqaa-MuiTypography-root"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Contact
            </p>
            <div
              className="css-d3iqhn-MuiStack-root"
            >
              <img
                className="css-1l0x7mi"
                src="facebook-icon.svg"
              />
              <img
                alt="twitter icon"
                className="css-1l0x7mi"
                src="twitter-icon.svg"
              />
              <img
                alt="gmail icon"
                className="css-1l0x7mi"
                src="gmail-icon.svg"
              />
              <img
                alt="linkedin icon"
                className="css-1l0x7mi"
                src="linkedin-icon.svg"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-vkfc0g-MuiGrid-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
          >
            Copyright ©2022 All rights reserved | by ATLP-cabal
          </p>
        </div>
      </div>
    </div>
  </div>,
]
`;
